src/actions/counterValues.js

```diff
@@ -22,35 +24,30 @@ export const updateCounterValues: UpdateCounterValues = payload => ({
   payload
 });
 
-export type FetchCounterValuesHist = (
-  ?number
-) => (Dispatch<*>, Function) => Promise<any>;
+type PollRates = () => (Dispatch<*>, () => State) => Promise<*>;
 
-export const fetchCounterValuesHist: FetchCounterValuesHist = () => async (
-  dispatch,
-  getState
-) => {
-  const { accounts, settings } = getState();
-  const { counterValue } = settings;
+// TODO pollRates will be abstracted out in wallet-common
+// because it will implement our draft proposal for the next API
+export const pollRates: PollRates = () => async (dispatch, getState) => {
+  const state = getState();
+  const { accounts, counterValues } = state;
+  const fiatUnit = fiatUnitSelector(state);
   const currencies = [...new Set(accounts.map(a => a.currency))];
-
-  const res = await fetchHistodayRates(currencies, getFiatUnit(counterValue));
-  if (Object.keys(res).length !== 0) {
-    dispatch(updateCounterValues(res));
-  }
-  return res;
-};
-
-export type FetchCounterValuesLatest = () => (Dispatch<*>, Function) => void;
-
-export const fetchCounterValuesLatest: FetchCounterValuesLatest = () => (
-  dispatch,
-  getState
-) => {
-  const { accounts, settings } = getState();
-  const { counterValue } = settings;
-  const currencies = [...new Set(accounts.map(a => a.currency))];
-  fetchCurrentRates(currencies, getFiatUnit(counterValue)).then(data => {
-    dispatch(updateCounterValues(data));
-  });
+  const getLatestDayFetched = cur => {
+    const all = {
+      ...(counterValues[cur.ticker] || {})[fiatUnit.ticker]
+    };
+    delete all.latest;
```

Comment:
consider using [_.omit](http://devdocs.io/lodash~4/index#omit) to avoid creating slow obj

Reply:
how is it faster? 🤔

Reply:
v8 will less likely optimize property access if some properties are added/deleted after object creation ([more details](https://v8project.blogspot.fr/2017/08/fast-properties.html), [bench](https://jsperf.com/delete-vs-undefined-vs-null/16))

Reply:
ok interesting. it's temporary code anyway, later we will have the latestDay as part of the meta data that our future api will return (see latest draft) so it will be `O(1)`